react-custom-hooks/client/src/components/useTodos.js

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * Manages the Todos by reading from and writing to the backend API.
+ *
+ * @returns {Object} with the following properties:
+ *   isLoading: a boolean indicating if the data has not yet loaded;
+ *   error: any error that occurred
+ *   todos: a list of To Do items;
+ *   addTodo: a function (newTodo) that will add a new To Do;
+ *   toggleCompleted: a function (todoId) that will toggle the completion of an existing To Do.
+ */
+export function useTodos() {
+  const [todos, setTodos] = useState();
+  const [error, setError] = useState();
+  useEffect(() => {
+    if (!todos) {
+      async function get() {
+        const bago = await readTodos();
+        setTodos(bago);
+      }
+      get();
+    }
+  }, [todos]);
+
+  async function addTodo(newTodo) {
+    setTodos([...todos, await createTodo(newTodo)]);
+  }
+
+  async function toggleCompleted(todoId) {
+    const index = todos.findIndex((el) => el.todoId === todoId);
```

Comment:
It's easier if you use `find` instead of `findIndex`. I use `find` about 10x more often than I use `findIndex`.